### PR TITLE
Autocomplete Update

### PR DESCRIPTION
**Pull request by bowenjw**
### Changes
- Added support for autocomplete
- Added example for autocomplete
- Added example for models
- Added example of string select menu
- Updated all modules to latest version
- Updated config handling in ExtendedClient
- Added Exstended client to discord.js objects see [PR ](https://github.com/bowenjw/typescript-discord-bot/pull/3)for more info 

**_Note: Documentation needs to be updated for changes, and the_** `de.ftl` **_need to be updated to match_** `us_en.ftl`**_._**

## Autocomplete
Autocomplete is now an optional function include in the `ChatInputCommand` interface
```ts
export interface ChatInputCommand extends Command {
    options: SlashCommandBuilder | SlashCommandSubcommandsOnlyBuilder | Omit<SlashCommandBuilder, 'addSubcommand' | 'addSubcommandGroup'>
    execute(client: ExtendedClient, interaction: ChatInputCommandInteraction): Promise<void>,
    autocomplete?(interaction: AutocompleteInteraction): Promise<void>
}
```
When an autocomplete interaction is received, the `ChatInputCommand` object is validated that it has the `autocomplete` function
```ts
switch (interaction.type) {
case InteractionType.ApplicationCommandAutocomplete:
    // Check if autocomplete interactions are enabled
    if (!client.config.interactions.receiveAutocomplete) return;
    interactionName = interaction.commandName;
    // eslint-disable-next-line no-case-declarations
    const autocomplete = client.commands.get(interactionName)?.autocomplete;
    if (!autocomplete) { console.warn(`[Warning] Autocomplete for ${interactionName} was not Setup`); }
    else { autocomplete(interaction); }
    break;
}
```

See `src/commands/example/autocomplete.ts` for example of auto complete

## Modal Example
Chat command creates a model window with a `short` and `paragraph` fields
image
Then replies with an embed containing the text
image
## Select Menu Example
Chat command crates select menu and updates it when selection in made

## Module Updates
[TypeScript]( https://devblogs.microsoft.com/typescript/announcing-typescript-5-0/) `4.9.4` -> `5.0.2` 
[discord.js](https://github.com/discordjs/discord.js/blob/main/packages/discord.js/CHANGELOG.md) `14.7.1` -> `14.8.0`
[@fluent/bundle](https://github.com/projectfluent/fluent.js/blob/master/fluent-bundle/CHANGELOG.md) `0.17.1` -> `0.18.0`
See `Package.json` for all changes

### CODE REVIEW

**Comment by bowenjw** [comment]
I have overhauled the interfaces and replaced them with classes.

File: src/interactions/modals/example/model.ts
Comment (Codeize) [inline src/interactions/modals/example/model.ts:1]: Hi, due to the `modal` word being misspelled in the `.setName` method here it causes the modal submit to fail without throwing any error.

It should be:
```diff
- .setName('model')
+ .setName('modal')
```

**Review by Codeize — CHANGES_REQUESTED** [review]

File: src/interactions/modals/example/model.ts
Comment (bowenjw) [reply src/interactions/modals/example/model.ts:1]: fixed

**Review by Codeize — APPROVED** [review]

**Comment by bowenjw** [comment]
Closing pull request as I am because I am working on a different one. Which will will have far exceeded what is in this pull request.

**Pull request closed without merging**